ribs/optimizers/_optimizer.py

```diff
@@ -94,6 +97,26 @@ def emitters(self):
         in this optimizer."""
         return self._emitters
 
+    def ask_dqd(self):
```

Comment:
To avoid duplicating code, make an `_ask_internal` and `_tell_internal` function that handles the shared code between `ask` and `ask_dqd`. Handle the logic for forcing ask-tell order in the respective functions though.